client/src/modules/user/UserProfile/DeleteAccount.jsx

```diff
@@ -0,0 +1,169 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, FormattedMessage } from 'react-intl';
+import { connect } from 'react-redux';
+import _flowRight from 'lodash/flowRight';
+import _trim from 'lodash/trim';
+
+import { IntlPropType } from 'common/constants/propTypes';
+import { deleteAccount, logoutCurrentUser } from '../model/actions';
+import './DeleteAccount.scss';
+import DeleteDialog from './DeleteDialog';
+import Dialog from 'common/components/Dialog';
+import AlertBox from 'common/components/AlertBox';
+import { MessageType } from 'common/constants/enums';
+
+class DeleteAccount extends Component {
+  state = {
+    email: '',
+    isDeleteDialogVisible: false,
+    isErrorVisible: false,
+    password: '',
+    pending: false,
+    verificationText: ''
+  };
+
+  handleDeleteAccount = async event => {
+    this.setState({ pending: true });
+    event.preventDefault();
+
+    const {
+      intl: { formatMessage }
+    } = this.props;
+    const { verificationText } = this.state;
+    const verificationString = formatMessage({ id: 'delete-form.verify-text' });
+
+    try {
+      if (verificationText === verificationString) {
+        const { email, password } = this.state;
+        const trimmedEmail = _trim(email);
+        const result = await deleteAccount(trimmedEmail, password);
+
+        this.setState({ isErrorVisible: false });
+
+        if (result) {
+          this.setState({
+            isAccountDeletedDialogVisible: true,
+            isDeleteDialogVisible: false,
+            pending: false
+          });
+        }
+      }
+
+      throw new Error();
```

Comment:
it throws error only when 
``` if (verificationText === verificationString) { ``` is false 
otherwise it goes inside the if and it never execute the error because deleteDialog is hidden after result is SUCCESS. 
but to be sure nothing unexpected happens I've added return within this if statement.